subprojects/testing-jvm/src/main/java/org/gradle/api/tasks/testing/Test.java

```diff
@@ -1014,7 +1019,18 @@ public File getBinResultsDir() {
      */
     @Incubating
     public void setBinResultsDir(File binResultsDir) {
-        this.binResultsDir = binResultsDir;
+        this.binResultsDir.set(binResultsDir);
+    }
+
+    /**
+     * Sets the root folder for the test results in internal binary format.
+     *
+     * @param binResultsDir The provider of the root folder
+     * @since 4.2
+     */
+    @Incubating
+    public void setBinResultsDir(Provider<? extends Directory> binResultsDir) {
```

Comment:
Let's not change the `Test` task for now. Instead, we should change the `XcTest` task so that it reuses a bunch of stuff from this but without extending it. We can do this as a later PR. For now, just use a hard-coded `File` to set the results directory instead of using a `Provider`.